setup.cfg

```diff
@@ -44,7 +44,10 @@ install_requires =
 	colorspacious>=1.1.2
 	pyqtgraph>=0.12.3
 	napari-ome-zarr>=0.3.2 # drag and drop convenience
-	napari[all] # uses pip-installed PyQt5 by default
+	napari[all];platform_system=='Windows' # uses pip-installed PyQt5 by default
+	napari[all];platform_system=='Linux' # uses pip-installed PyQt5 by default
+	napari;platform_system=='Darwin' # use PyQt6 from conda-forge
```

Comment:
How would we distinguish Intel-based and M-series based MacOS in this way?

Reply:
The `platform_machine` keyword might help. Ref: https://peps.python.org/pep-0508/#environment-markers

E.g. on my M1 mac:

```python_repl
>>> import platform
>>> platform.machine()
'arm64'
```

Reply:
It does make sense to use `platform_machine` keyword. 
Quoting from the ref: "The marker language is inspired by Python itself, chosen for the ability to safely evaluate it without running arbitrary code that could become a security vulnerability. " 

Can you try, `napari;platform_system=='Darwin' and platform_machine=='arm64'` and see if that works?

Reply:
I tried with:

```config
	napari[all];'arm64' not in platform_machine # with Qt5 and skimage
	napari;'arm64' in platform_machine # without Qt and skimage
	PyQt6;'arm64' in platform_machine
```

And it seems to work on an M1 mac. Still need to test with an Intel mac to confirm (because I'm not sure which arch are the GH test runners using).

![image](https://user-images.githubusercontent.com/67518483/217389262-7b3ed6ae-dfd9-4dc2-b7f4-0b4c1a24b438.png)

Reply:
At @ziw-liu's request, I just tested an installation on my intel mac---everything seems to be working well (see attached screenshot)
<img width="1440" alt="Screen Shot 2023-02-07 at 3 35 02 PM" src="https://user-images.githubusercontent.com/9554101/217391420-a3709421-6715-43ab-b96b-daf6cc43cc90.png">

I also tested on my biohub M1 mac (but with Rosetta I think?). Again, everything seems to be working. 
<img width="1352" alt="Screen Shot 2023-02-07 at 3 43 20 PM" src="https://user-images.githubusercontent.com/9554101/217392373-91e842c3-5106-4d67-ae9f-1995b261bbdf.png">

Thanks for digging up the `platform_system` keywords @mattersoflight !

Reply:
@talonchandler Thanks for help test this!